### PR TITLE
feat: api v3 uses s3_uri 

### PR DESCRIPTION
**Pull request by MDunitz**
#### Reviewers
**Functional:** @atolopko-czi 

**Readability:** @danieljhegeman 

---

## Changes
- add v0.3 API code. v0.3 uses the path /s3_uri/{s3_uri}/api/v0.3/. 
   - /config
   - /colors/
   - /annotations/obs
   - /annotations/var
   - /data/var
   - /genesets
   - /layout/obs
   - /schema
   - /summarize/var
- Add /{data_root}/{dataset_uuid}/api/v0.3/metadata 
- Add /{data_root}/{dataset_uuid}/api/v0.3/s3_uri
- To the s3_uri in the URL path is must be made URL safe. 
- move v0.2 API code to its own file
- move cache controls to its own file

Related to #167

### CODE REVIEW

File: server/app/api/v2.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
+import logging
```
Comment (Bento007) [inline server/app/api/v2.py:1]: This code is extracted from server/app/app.py

File: server/app/api/__init__.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+from functools import wraps
```
Comment (Bento007) [inline server/app/api/__init__.py:1]: This code was moved from server/app/app.py to here.

File: hosted/create_rdev_config.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+  diffexp:
+    enable: true
+    lfc_cutoff: 0.01
+    top_n: 10
+"""
+
+if __name__ == "__main__":
+    with open("config.yaml", "w") as f:
+        f.write(config)
```
Comment (danieljhegeman) [inline hosted/create_rdev_config.py:146]: probably add back in the newline, yeah?

**Review by danieljhegeman — APPROVED** [review]

File: server/app/api/v2.py [hunk 4]
```diff
@@ -6,38 +6,38 @@
...
 
 
 def rest_get_data_adaptor(func):
     @wraps(func)
-    def wrapped_function(self, s3_uri=None):
+    def wrapped_function(self, dataset=None):
         try:
-            s3_uri = unquote(s3_uri) if s3_uri else s3_uri
-            data_adaptor = get_data_adaptor(s3_uri)
+            dataset = unquote(dataset) if dataset else dataset
```
Comment (Bento007) [inline server/app/api/v2.py:28]: This is not needed if dataset is passed in.

Comment (Bento007) [reply server/app/api/v2.py:28]:
```suggestion
```

**Review by Bento007 — APPROVED** [review]

**Comment by atolopko-czi** [comment]
This PR's changes were included in and superseded by #194

**Pull request closed without merging**